### DISCUSSION TITLE
get current menuPlacement when set to auto

### DISCUSSION THREAD

**Discussion by amsuarez99** [JedWatson/react-select#4872] [Q&A]
I want to define a custom set of styles based on the menuPlacement when it is set to auto. Currently, I found that the only prop related to the menu is `isMenuOpen`. How can I get the specified behaviour, am I missing something?

**Comment by vjee** [comment]
Are you trying to style it via CSS in JS? Have you tried https://react-select.com/styles?
If you are using plain css, you could add a classname to the select in the case you set `menuPlacement` to `"auto"`. That way you can target that class with a CSS selector.

**Reply by ebonow** [reply]
To add to this, the "state" prop passed thru the styles api should contain the selectProps so you could style it specifically if the prop value is `auto`

```jsx 
const styles = {
  menu: (provided, state) => ({ 
    ...provided,
    ...(state.selectProps.menuPlacement === 'auto' ? { outline: '1px dotted red' }  : {})
  })
};

<Select styles={styles} menuPlacement="auto" {...otherProps} />
```
